compiler/resolution/cullOverReferences.cpp

```diff
@@ -243,7 +243,8 @@ static
 bool symExprIsSet(SymExpr* se)
 {
   // The ref is necessary if it is for an explicit ref var
-  if (se->symbol()->hasFlag(FLAG_REF_VAR)) {
+  if (se->symbol()->hasFlag(FLAG_REF_VAR)                             &&
+      ! se->symbol()->hasFlag(FLAG_DEFAULT_INTENT_IS_REF_MAYBE_CONST) ) {
```

Comment:
This is fine, but I wouldn't have expected your shadow variables to have FLAG_REF_VAR set.